api/kyverno/v2alpha1/policy_exception_types.go

```diff
@@ -51,16 +51,32 @@ func (p *PolicyException) Contains(policy string, rule string) bool {
 
 // PolicyExceptionSpec stores policy exception spec
 type PolicyExceptionSpec struct {
+	// Background controls if exceptions are applied to existing policies during a background scan.
+	// Optional. Default value is "true". The value must be set to "false" if the policy rule
+	// uses variables that are only available in the admission review request (e.g. user name).
+	Background *bool `json:"background,omitempty" yaml:"background,omitempty"`
+
 	// Match defines match clause used to check if a resource applies to the exception
 	Match kyvernov2beta1.MatchResources `json:"match"`
 
 	// Exceptions is a list policy/rules to be excluded
 	Exceptions []Exception `json:"exceptions"`
 }
 
+func (p *PolicyExceptionSpec) BackgroundProcessingEnabled() bool {
+	if p.Background == nil {
+		return true
+	}
+	return *p.Background
+}
+
 // Validate implements programmatic validation
 func (p *PolicyExceptionSpec) Validate(path *field.Path) (errs field.ErrorList) {
-	errs = append(errs, p.Match.Validate(path.Child("match"), false, nil)...)
+	if p.BackgroundProcessingEnabled() {
+		errs = append(errs, p.Match.ValidateResourceWithNoUserInfo(path.Child("match"), false, nil)...)
```

Comment:
Same here, why not have `ValidateResourceWithNoUserInfo` added inside `p.Match.Validate()`?

Reply:
that's right

Reply:
@realshuting i did this becaues `p.Match.Validate()` calls `userInfo.Validate()`. it doesn't validate that there should be no userInfo. An option might be to move userInfo validation out of validate.